src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Indentation/CSharpIndentationService.Indenter.cs

```diff
@@ -299,45 +299,57 @@ private static IndentationResult GetIndentationBasedOnToken(Indenter indenter, S
 
             case SyntaxKind.CommaToken:
                 {
-                    return GetIndentationFromCommaSeparatedList(indenter, token);
+                    if (TryGetIndentationFromCommaSeparatedList(indenter, token) is { } result)
+                    {
+                        return result;
+                    }
+
+                    break;
                 }
 
             case SyntaxKind.CloseParenToken:
                 {
                     if (token.Parent.IsKind(SyntaxKind.ArgumentList))
                     {
-                        return GetDefaultIndentationFromToken(indenter, token.Parent.GetFirstToken(includeZeroWidth: true));
+                        token = token.Parent.GetFirstToken(includeZeroWidth: true);
                     }
 
-                    goto default;
+                    break;
                 }
+        }
 
-            default:
-                {
-                    return GetDefaultIndentationFromToken(indenter, token);
-                }
+        var result = TryGetIndentationForQueryExpression(indenter, token)
+            ?? TryGetIndentationFromMultilineStatement(indenter, token);
+
+        if (result is not null)
+        {
+            return result.Value;
         }
+
+        var spaceToAdd = indenter.Options.FormattingOptions.IndentationSize;
+
+        return indenter.IndentFromStartOfLine(indenter.Finder.GetIndentationOfCurrentPosition(indenter.Tree, token, position, spaceToAdd, indenter.CancellationToken));
```

Comment:
i would mind documenting the prior 10 lines.  just to somewhat explain what's going on.

Reply:
I pushed some updates and added a bit more documentation. Lmk if there's more you'd like to see. I don't know the explanation for why TryGetIndentationForQueryExpression and TryGetIndentationFromMultilineStatement are done in this order. This ordering has just been preserved and made easier to see.